src/block/stdio_executor.rs

```diff
@@ -140,7 +143,8 @@ impl<B: Backend> StdIoBackend<B> {
     /// # Arguments
     /// * `inner` - The block device backend.
     /// * `features` - The features that were negotiated between driver and device.
-    pub fn new(mut inner: B, features: u64) -> Result<Self> {
+    /// * `device_id` - The block device id.
+    pub fn new(mut inner: B, features: u64, device_id: [u8; VIRTIO_BLK_ID_BYTES]) -> Result<Self> {
```

Comment:
Does the`GET_ID` operation always have to be supported? If not, does it make sense to have the `device_id` field as an `Option<...>` and return unsupported when it's `None`? In that case, we could also add a separate constructor (something like `with_device_id`), and keep the old `new` signature.